core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -96,6 +96,15 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
 
   def leftMap[C](f: A => C)(implicit F: Functor[F]): EitherT[F, C, B] = bimap(f, identity)
 
+  def leftFlatMap[BB >: B, D](f: A => EitherT[F, D, BB])(implicit F: Monad[F]): EitherT[F, D, BB] =
+    EitherT(F.flatMap(value) {
+      case Left(a) => f(a).value
+      case r@Right(_) => F.pure(r.leftCast)
+    })
+
+  def leftSemiflatMap[D](f: A => F[D])(implicit F: Monad[F]): EitherT[F, D, B] =
+    leftFlatMap(a => EitherT.left(f(a)))
```

Comment:
this is going to allocate the `EitherT` just to call `.value` above. Why not just inline the implementation:
```scala
EitherT(F.flatMap(value) {
  case Left(a) => f(a).map { d => Left(d) }
  case r@Right(_) => F.pure(r.leftCast)
})
```

Reply:
Should I rewrite `leftFlatMap` in terms of `leftSemiflatMap` then?